packages/next/src/client/components/navigation-devtools.ts

```diff
@@ -0,0 +1,179 @@
+import type { FlightRouterState } from '../../shared/lib/app-router-types'
+import type { Params } from '../../server/request/params'
+import {
+  createDevToolsInstrumentedPromise,
+  type InstrumentedPromise,
+  type NavigationPromises,
+} from '../../shared/lib/hooks-client-context.shared-runtime'
+import {
+  computeSelectedLayoutSegment,
+  getSelectedLayoutSegmentPath,
+} from '../../shared/lib/segment'
+import { ReadonlyURLSearchParams } from './navigation.react-server'
```

Comment:
This still scares me. I assume `ReadonlyURLSearchParams` is not actually something React Server related? But then it still enters a React Server module graph when we're actually in React Client code here.

Reply:
yeah - going to pull this out into it's own module